### PR TITLE
Gate release action to rust-analyzer owner

### PR DESCRIPTION
**Pull request by swarnimarun**
Should Fix https://github.com/rust-analyzer/rust-analyzer/issues/3589

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ jobs:
   dist:
     name: dist
     runs-on: ${{ matrix.os }}
+    if: github.repository == "rust-analyzer/rust-analzyer"
```
Comment (flodiebold) [inline .github/workflows/release.yaml]:
```suggestion
    if: github.repository == "rust-analyzer/rust-analyzer"
```

Comment (swarnimarun) [reply .github/workflows/release.yaml]: LMAO....
How did I mis type that....

**Comment by matklad** [comment]
Thanks @swarnimarun and @flodiebold !

**Comment by DianaNites** [comment]
FYI this PR broke releases, https://github.com/rust-analyzer/rust-analyzer/runs/515231288?check_suite_focus=true

Noticed because the nightly is 2 days old

**Pull request merged**